lib/crc/crc24_sw.c

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright (C) 2024 BayLibre.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/sys/crc.h>
+
+#define CRC24_POLY 0x01864cfbU
```

Comment:
Pedantry: wikipedia doesn't list this polynomial or an obvious inversion/reversal permutation as a "crc24".  It looks like this is specific to OpenPGP?  If so, might be worth naming it that way (e.g. "crc24_pgp", or "crc24_rfc4880") to avoid collisions in the future.

Reply:
Yes, it seems to be specific to OpenPGP.
There are also CRC-24 for the FlexRay bus, so it is a good idea to use a more precise name.
I renamed what relevant as `crc24_pgp`.